### ISSUE TITLE
tf.keras.layers.BatchNormalization produces wrong results when original feature scale is moderately small.

### ISSUE THREAD

**Issue by cottrell** [tensorflow/tensorflow#33027]
It looks like BatchNormalization fails to scale features up if the original scale is too low ... or I am doing something brain dead here. 

If this is actually a constraint on the scale param, it should be very explicitly documented. 

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): YES
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0.0rc2
- Python version: 3.7.3
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

```
import tensorflow as tf
import tensorflow_probability as tfp

x = np.random.randn(1000, 1).astype(np.float32)

x = x * 0.001

bn = tf.keras.layers.BatchNormalization()

@tf.function
def _run_one(x):
    bn(x, training=True)

def run(x):
    for i in range(10000):
        _run_one(x)

tf.print(tf.math.reduce_std(bn(x, training=False), axis=0), tf.reduce_mean(bn(x, training=False), axis=0))
run(x)
tf.print(tf.math.reduce_std(bn(x, training=False), axis=0), tf.reduce_mean(bn(x, training=False), axis=0))

[0.000992934452] [-2.47328098e-05]
[0.0313995518] [-2.12341544e-09]
```

**Describe the expected behavior**
```

import tensorflow as tf
import tensorflow_probability as tfp

x = np.random.randn(1000, 1).astype(np.float32)

# x = x * 0.001
x = x * 2

bn = tf.keras.layers.BatchNormalization()

@tf.function
def _run_one(x):
    bn(x, training=True)

def run(x):
    for i in range(10000):
        _run_one(x)

tf.print(tf.math.reduce_std(bn(x, training=False), axis=0), tf.reduce_mean(bn(x, training=False), axis=0))
run(x)
tf.print(tf.math.reduce_std(bn(x, training=False), axis=0), tf.reduce_mean(bn(x, training=False), axis=0))


[2.03433657] [0.100093901]
[0.999882281] [1.81198118e-07]
```

**Code to reproduce the issue**

See above.

**Other info / logs**

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.0.0rc2 and was able to reproduce the issue.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/9dd7984692101e09e5c9b816b1a3643a/untitled245.ipynb).Thanks!

**Comment by ppwwyyxx** [comment]
This is not a bug but expected.
Using `BatchNormalization(epsilon=0.0)` will make it print your expected output.

**Comment by cottrell** [comment]
> This is not a bug but expected.
> Using `BatchNormalization(epsilon=0.0)` will make it print your expected output.


Seems correct.

    epsilon=0.001,

This default is way too low and we should get a warning added to the docs ... the default should be much, much lower.

**Comment by jvishnuvardhan** [comment]
@cottrell Would you like to create any PR to update the docs? Thanks!

**Comment by cottrell** [comment]
Seems like the incentives are aligned to leave it as is. But let's leave this issue here for now. I might follow up with some digging on the source of this bad-value attack.

**Comment by cottrell** [comment]
PR something but I likely have to read more about how to do this. Am hoping the PR triggers some tests that I can then chase later. Let me know if this is bad.

**Comment by robieta** [comment]
Closing as intended behavior.
